### DISCUSSION TITLE
Would this SDK work for the tag in Adidas GMR insole?

### DISCUSSION THREAD

**Discussion by KennyKYLam** [google/JacquardSDKiOS#5] [Q&A] [answered]
Question in the title, it would be cool to play with that data. If not, will it be supported later?

**Comment by gald89** [comment]
I was thinking the same thing. How similar is the Jacquard 2.0 to the previous model?

**Comment by aufflick** [comment]
Hi Kenny, the answer is yes and no :) Yes it will connect to the tag you get with an Adidas GMR insole, but as of now it won't do all that much. You can blink the LED and buzz the haptics, the other Jacquard-fabric-specific functionality will unsurprisingly not do anything. Keep a watch on the repository for future updates that may be more applicable :)

**Reply by KennyKYLam** [reply]
Can't wait for the gyrometer or accelerometer stuff to come! :)

**Comment by aufflick** [comment, marked as answer]
We just released version 0.2.0 of the SDK which includes the first part of IMU access. This release includes the ability to record and then download IMU data from the tag which might be interesting to GMR users.

The SDK is now also open source :)

Note that this is raw IMU data and does not include any interpretation (eg. like the GMR app interprets kicks etc - the machine learning models that do this are specific to the Adidas product).

Using this IMU function requires the SDK to update the firmware on the tag and install a new loadable module replacing the GMR module. It's possible that the GMR app can restore original GMR functionality after this has happened but it's not guaranteed - if you want to use both please feel free to file a bug and we'll investigate how we can make this robust.

**Reply by KennyKYLam** [reply]
Can I use this app https://apps.apple.com/us/app/jacquard-by-google/id1204971157 to update firmware to work with 0.2.0? Just want to confirm.

**Reply by aufflick** [reply]
Yes you can use the App Store app to upgrade the firmware, just make sure that you either remove the tag from the app or delete the app afterwards (each tag can only connect to one installed app at once). This page should help with that: https://google.github.io/JacquardSDKiOS/updating-firmware.html

The latest version of the SDK does support updating firmware as well, but using the App Store app to update first is probably a good idea, since there are some edge cases in the SDK firmware updating that we have a fix on the way for.
